### PR TITLE
plugins/grug-far: init

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/2133

### CODE REVIEW

File: tests/test-sources/plugins/by-name/grug-far/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  empty = {
+    plugins.grug-far.enable = true;
+  };
+
+  # Just a small subset of the settings
+  defaults = {
+    plugins.grug-far = {
+      enable = true;
+      settings = {
```
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/grug-far/default.nix:10]:
```suggestion
      enable = true;

      settings = {
```

@GaetanLepage would make this suggestion. Personally I don't care too much, but I guess we should be consistent :joy:

File: plugins/by-name/grug-far/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    searchOnInsertLeave = true;
+    maxWorkers = 8;
+    engine = "ripgrep";
+    engines = {
+      ripgrep = {
+        path = "rg";
+        showReplaceDiff = true;
+      };
+    };
+  };
```
Comment (MattSturgeon) [inline plugins/by-name/grug-far/default.nix:23]: I'm assuming it's intentional to have this (excellent) settings example and the defaults test case, but not actually declare any settings sub-options?

Not complaining! Just double checking. :heart_eyes:

File: tests/test-sources/plugins/by-name/grug-far/default.nix
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/grug-far/default.nix:1]: Would there be any value in also having an `example` test case based on the settingsExample? Or is that a waste of resources?

File: plugins/by-name/grug-far/default.nix (hunk 2 above)
Comment (khaneliman) [reply plugins/by-name/grug-far/default.nix:23]: I just saw the wealth of options they had and don't have the energy to document any atm until I spend any time with the plugin and see what's really useful.

Comment (MattSturgeon) [reply plugins/by-name/grug-far/default.nix:23]: I actually prefer not having them as nix options, especially if it seems they may be difficult to understand/maintain.

Comment (traxys) [reply plugins/by-name/grug-far/default.nix:23]: My heuristic is to look at the readme, if they explicitly mention ~5 options I'd add them, else I would not add anything

Comment (GaetanLepage) [reply plugins/by-name/grug-far/default.nix:23]: > I'm assuming it's intentional to have this (excellent) settings example and the defaults test case, but not actually declare any settings sub-options?

I see myself in this picture and I don't like it.

File: tests/test-sources/plugins/by-name/grug-far/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply tests/test-sources/plugins/by-name/grug-far/default.nix:10]: > @GaetanLepage would make this suggestion.

I would indeed.
ADD THIS SPACE.

NOW.
Or I'm blocking this PR

Comment (GaetanLepage) [reply tests/test-sources/plugins/by-name/grug-far/default.nix:10]: (And please, declare all the 420 options of this plugin with proper types and sub-element-rawtype-support, or this doesn't get merged.)

**Review by traxys — APPROVED** [review]

**Comment by khaneliman** [comment]
@mergify queue

**Pull request merged**